src/conversion.rs

```diff
@@ -347,6 +340,45 @@ where
     }
 }
 
+mod into_pyobject_ext {
+    pub trait Sealed {}
+    impl<'py, T> Sealed for T where T: super::IntoPyObject<'py> {}
+}
+
+/// Convenience methods for common usages of [`IntoPyObject`]. Every type that implements
+/// [`IntoPyObject`] also implements this trait.
+///
+/// These methods:
+///   - Drop type information from the output, returning a `PyAny` object.
+///   - Always convert the `Error` type to `PyErr`, which may incur a performance penalty but it
+///     more convenient in contexts where the `?` operator would produce a `PyErr` anyway.
+pub trait IntoPyObjectExt<'py>: IntoPyObject<'py> + into_pyobject_ext::Sealed {
+    /// Converts `self` into an owned Python object, dropping type information.
+    #[inline]
+    fn into_py_any(self, py: Python<'py>) -> PyResult<Bound<'py, PyAny>> {
+        self.into_pyobject(py)
+            .map(|obj| obj.into_any().into_bound())
+            .map_err(Into::into)
+    }
+
+    /// Converts `self` into a Python object, dropping type information.
+    ///
+    /// This is typically only useful when the resulting output is going to be passed
+    /// to another function that only needs to borrow the output.
+    #[inline]
+    fn into_bound_object_py_any(
```

Comment:
The name of this method is open to bikeshedding, I think I'm happy with this but there's probably a better one.